tests/targets_tests.py

```diff
@@ -16,6 +18,66 @@
 class TargetsTests(TestCase):
 
     @parameterized.expand([
+        (luigi.format.Gzip, b'x,y', b'\x1f\x8b'),  # gz magic number
+        (None, 'x,y', b'x,'),  # first two chars of csv header
+    ])
+    def test_compressed_is_compressed_FireflowerS3Target(self, format, contents, first_two_bytes):
+        with TempDirectory() as d:
+
+            def getenv(v, _):
+                # Is there a better way to assert arguments?
+                self.assertEqual(v, 'LOCAL_S3_PATH')
+                return d.path
+
+            with mock.patch('fireflower.targets.os.getenv',
+                            side_effect=getenv):
+                s = FireflowerS3Target('s3://test.csv.gz', format=format)
+                with s.open('w') as f:
+                    f.write(contents)
+                with open(os.path.join(d.path, 'test.csv.gz'), 'rb') as f:
+                    assert f.read(2) == first_two_bytes
+
+    @parameterized.expand([
+        (True, b'\x1f\x8b'),  # gz magic number
+        (False, b'x,'),  # first two chars of csv header
+    ])
+    def test_compressed_is_compressed_S3CSVTarget(self, compressed, first_two_bytes):
+        with TempDirectory() as d:
+
+            def getenv(v, _):
+                # Is there a better way to assert arguments?
+                self.assertEqual(v, 'LOCAL_S3_PATH')
+                return d.path
+
+            with mock.patch('fireflower.targets.os.getenv',
+                            side_effect=getenv):
+                s = S3CSVTarget('s3://test.csv.gz', compressed=compressed)
+                tuples = [(1, 2), (3, 4)]
+                s.write_csv_tuples(tuples, ('x', 'y'))
+                with open(os.path.join(d.path, 'test.csv.gz'), 'rb') as f:
+                    assert f.read(2) == first_two_bytes
+
+    def test_local_file_passthru_bytes(self):
+        with TempDirectory() as d:
+
+            def getenv(v, _):
+                # Is there a better way to assert arguments?
+                self.assertEqual(v, 'LOCAL_S3_PATH')
+                return d.path
+
+            with mock.patch('fireflower.targets.os.getenv',
+                            side_effect=getenv):
+
+                # Passthru bytes with format=Nop instead of trying to encode/decode as utf8
+                s = FireflowerS3Target('some_file.txt.gz', format=luigi.format.Nop)
+                with s.open('w') as fout:
+                    fout.write(gzip.compress('asdf'.encode('utf8')))
+                    fout.write(gzip.compress('fdsa'.encode('utf8')))
+                with s.open('r') as fin:
+                    expected_bytes = fin.read()
```

Comment:
Ah, no, we aren't getting rid of compression functionality:
- FireflowerS3Target wasn't compressing or decompressing in the first place. It was only encoding/decoding text (TextIOWrapper) or not.
- Now that we're passing through luigi's `format` in both the s3 and local case, we can get compression using luigi `format`'s directly: http://luigi.readthedocs.io/en/stable/_modules/luigi/format.html
```py
Text = TextFormat()
UTF8 = TextFormat(encoding='utf8')
Nop = NopFormat()
SysNewLine = NewlineFormat()
Gzip = GzipFormat()
Bzip2 = Bzip2Format()
MixedUnicodeBytes = MixedUnicodeBytesFormat()
```